### PR TITLE
fix: use correct environment file for rendering spa fallback page

### LINKED ISSUE #10801: Sveltekit always writes production env variables inside index.html
### Describe the bug

I am really struggeling to get the `mode` flag to work for different builds for different environments (dev, staging, test, production). 
The build always ends up with using the variables from the `env.production` file.

### Reproduction

I have the following structure:

.env-file
```
PUBLIC_API=env
```

.env.staging
```
PUBLIC_API=staging
```

.env.production
```
PUBLIC_API=production
```

package.json

```
"build:staging": "vite build --mode staging",
```

### Logs

```Shell
No matter what if I run the command `npm run build:staging` vite is telling me that it is using the mode flag: `vite v4.3.9 building for staging...`
```

### System Info

```Shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 9.53 GB / 31.95 GB
  Binaries:
    Node: 16.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.19.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (117.0.2045.43)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @sveltejs/adapter-static: ^2.0.2 => 2.0.2
    @sveltejs/kit: ^1.20.0 => 1.20.0
    svelte: ^3.59.1 => 3.59.1
    vite: ^4.3.9 => 4.3.9
```

### Severity

annoyance

### Additional Information

 I always get the following output inside the generated index.html file:

```html
__sveltekit_1wnehg = {
    base: "",
    env: { "PUBLIC_API": "production" }
};
```

It is ignoring the mode flag completely inside the build. 

Any idea what I am missing here or is this indeed a bug? 

UPDATE:

**The build is using indeed after double checking the correct environment variables inside the SPA**. But why are there the env variables for the production environment always written to the index.html if the have no effect?

UPDATE 2:

**I checked the build files and found out about the `public.js` (inside the server folder)**, which is indeed using the correct env variables.

```
const PUBLIC_API="staging";
export {
  PUBLIC_API as P,
};
```

When I search through the generated js files I can indeed only find the env variables from the correct build mode!

### PR DESCRIPTION
**Pull request by MarkusWendorf**
fixes #10801 

The `adapter-static` fallback page is hardcoded to use `production` environment variables (this is defined in `kit` and called from `adapter-static`). This PR uses the correct environment specified by `vite_config.mode`.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by eltigerchino — CHANGES_REQUESTED** [review]
Could you move the test to `adapter-static/test/apps/spa`? I think the fallback page test fits better there since it already has one among the other SPA tests. We can also extend the `+error.svelte` file to display the environment variable value  then navigate to a non-existent route to see the value.

Other than that, it looks really good to me! Thank you!

File: packages/adapter-static/test/apps/spa/test/test.js [hunk 1]
```diff
@@ -27,3 +27,9 @@ test('renders error page for missing page', async ({ page }) => {
 	await page.goto('/nosuchpage');
 	expect(await page.textContent('h1')).toEqual('404');
 });
+
+test('uses correct environment variables for fallback page (mode = staging)', async ({ page }) => {
+	await page.goto('/');
```
Comment (eltigerchino) [inline packages/adapter-static/test/apps/spa/test/test.js:32]:
```suggestion
```

File: packages/adapter-static/test/apps/spa/src/routes/fallback/[...rest]/+page.svelte [hunk 2]
```diff
@@ -1 +1,7 @@
+<script>
+	import { env } from '$env/dynamic/public';
+</script>
+
 <h1>the fallback page was rendered</h1>
+
+<b>{env.PUBLIC_VALUE}</b>
```
[the remark this answers is not in the document]
Comment (MarkusWendorf) [reply packages/adapter-static/test/apps/spa/src/routes/fallback/[...rest]/+page.svelte:7]: This is a dynamic route that is not being pre-rendered and so the fallback is used. So this also has the issue with the environment variables being wrong.

File: packages/adapter-static/test/apps/spa/test/test.js [hunk 3]
```diff
@@ -27,3 +27,9 @@ test('renders error page for missing page', async ({ page }) => {
...
+	await page.goto('/fallback/x/y/z');
```
[the remark this answers is not in the document]
Comment (MarkusWendorf) [reply packages/adapter-static/test/apps/spa/test/test.js:32]: As mentioned above `/fallback/x/y/z` already uses the fallback and exhibits the issue this PR fixes.

[the remark this answers is not in the document]
Comment (eltigerchino) [reply packages/adapter-static/test/apps/spa/test/test.js:32]: Ah you’re absolutely right! Sorry, my understanding of the fallback page is still a bit fuzzy.

**Review by eltigerchino — APPROVED** [review]

**Comment by eltigerchino** [comment]
Thank you!

**Pull request merged**